solc/CommandLineInterface.cpp

```diff
@@ -658,7 +658,19 @@ bool CommandLineInterface::parseArguments(int _argc, char const* const* _argv)
 		return false;
 	}
 
-	parser.parse(_argc, _argv);
+	try
+	{
+		parser.parse(_argc, _argv);
+	}
+	catch (...)
+	{
+		// Even if the overall CLI parsing fails, the --color/--no-color options may have been
+		// successfully parsed, and if so, should be taken into account when printing errors.
+		// If no value is present, it's possible that --no-color is still there but parsing failed
+		// due to other, unrecognized options so play it safe and disable color in that case.
+		m_options.formatting.coloredOutput = parser.options().formatting.coloredOutput.value_or(false);
```

Comment:
Can't get this to work with unrecognized options, e.g. `solc --color --unrecognized-command` and `solc --unrecognized-command` will both be printed as plain text; not sure how the boost parser works under the hood, perhaps it first validates that all program options are valid, and only then stores values.

Reply:
Doesn't the comment explain that? It says that unrecognized options may result in us not seeing `--no-color` (and other options, including `--color`, as well) when parsing fails. We assume `--no-color` in that case for that reason.

I guess I could handle specifically the case of unrecognized options by exposing the `m_args` object containing the result of parsing by boost and checking if `--color` or `--no-color` are still there. I didn't think it was really worth complicating the mechanism though. First, I'm not sure if I can even assume that `m_args` is filled out if parsing fails (I'd have to check). Second, there are other kinds of parsing errors so this will never be perfect anyway. There will always be some more or less obscure corner cases where `--color` or `--no-color` gets ignored.

So the question is whether you think it's worth complicating things just for this case.

Reply:
ok, so that's what I thought too. It could be made to handle those corner cases but it's not worth it.